src/v/cluster/data_policy_manager.cc

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cluster/data_policy_manager.h"
+
+#include "cluster/cluster_utils.h"
+#include "cluster/errc.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/future.hh>
+
+#include <system_error>
+
+namespace cluster {
+
+namespace {
+
+template<typename Cmd>
+ss::future<std::error_code> do_apply(
+  ss::shard_id shard,
+  Cmd cmd,
+  ss::sharded<data_policy_manager::container_type>& db) {
+    return db.invoke_on(
+      shard,
+      [cmd = std::move(cmd)](
+        data_policy_manager::container_type& local_db) mutable {
+          return ss::visit(
+            std::move(cmd),
+            [&local_db](create_data_policy_cmd cmd) {
+                local_db.insert_or_assign(cmd.key, std::move(cmd.value.dp));
```

Comment:
`create` should return an error if it exists, and `update` (which we can add later) would return an error if it doesn't exist.